pkg/apis/vsphere/v1/template_helpers.go

```diff
@@ -0,0 +1,73 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+)
+
+var (
+	// ErrTemplateNotFound is returned when the named template was not found at a given location
+	ErrTemplateNotFound = fmt.Errorf("%w: named template was not found at specified location", api.ErrNotFound)
+)
+
+const (
+	// LatestTemplateBuild is used to find the template with the highest build number
+	LatestTemplateBuild = "latest"
+)
+
+// FindNamedTemplate retrieves a template by name and build at a specified location.
+// Empty and LatestTemplateBuild build identifier will yield the highest available build.
+// It returns ErrTemplateNotFound if no matching template was found.
+func FindNamedTemplate(ctx context.Context, a api.API, name, build string, location corev1.Location) (*Template, error) {
+	var (
+		match   *Template
+		buildNo = -1
+		latest  = build == "" || build == LatestTemplateBuild
```

Comment:
for those two you don't need `var` when just using `:=` and then it's only using var for `match`, so please change to
```go
var match *Template
buildNo := -1
// ...
```